metadata-ingestion/src/datahub/ingestion/source/gc/soft_deleted_entity_cleanup.py

```diff
@@ -335,16 +301,26 @@ def cleanup_soft_deleted_entities(self) -> None:
             return
         self.start_time = time.time()
 
-        futures: Dict[Future, str] = dict()
+        futures: Dict[Future, Urn] = dict()
         with ThreadPoolExecutor(max_workers=self.config.max_workers) as executor:
             for urn in self._get_urns():
+                try:
+                    self.report.num_soft_deleted_entity_found += 1
+                    soft_deleted_urn = Urn.from_string(urn)
+                except Exception as e:
```

Comment:
Will throw an invalid urn error - let’s catch the specific error type

Reply:
thanks, I fixed it.